### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by hkulekci**
Sample for https://github.com/appbaseio/mirage/issues/23 and also dockerize implementation for the dejavu project

### CODE REVIEW

**Comment by jalberto** [comment]
2 months but not news yet?

File: .docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+ENV APP_USER  "app"
+ENV APP_DIR "/project"
+
+RUN useradd --user-group --create-home --shell /bin/false $APP_USER
+
+WORKDIR $APP_DIR
+
+RUN npm install -g bower
+RUN npm install -g typescript
```
Comment (siddharthlatest) [inline .docker/Dockerfile]: Installing typescript is not necessary for dejavu. However, it does require installing gulp.

File: .docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ADD . $APP_DIR
+RUN chown -R $APP_USER:$APP_USER $APP_DIR
+USER $APP_USER
+
+RUN npm install
+RUN bower install
+
```
Comment (siddharthlatest) [inline .docker/Dockerfile]: After bower install, there should be a step for something along the lines of:

```RUN gulp compact```

Comment (hkulekci) [inline .docker/Dockerfile]: Added.

File: .docker/Dockerfile (hunk 1 above)
Comment (hkulekci) [inline .docker/Dockerfile]: I changed typescript as gulp

**Pull request merged**